### PR TITLE
poetry2nix: 1.14.0 -> 1.15.2

### PR DESCRIPTION
**Pull request by adisbladis**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix [hunk 1]
```diff
@@ -127,11 +127,23 @@ self: super:
...
+  datadog-lambda = super.datadog-lambda.overridePythonAttrs (old: {
+    postPatch = ''
+      substituteInPlace setup.py --replace "setuptools==" "setuptools>="
+    '';
+    buildInputs = old.buildInputs ++ [ self.setuptools ];
+  });
+
+  ddtrace = super.ddtrace.overridePythonAttrs (old: {
+    buildInputs = old.buildInputs ++
+      (pkgs.lib.optionals pkgs.stdenv.isDarwin [ pkgs.darwin.IOKit ]) ++ [ self.cython ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix:144]:
```suggestion
      (lib.optionals stdenv.isDarwin [ pkgs.darwin.apple_sdk.frameworks.IOKit ]) ++ [ self.cython ];
```

File: pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix [hunk 2]
```diff
@@ -245,7 +257,7 @@ self: super:
 
   horovod = super.horovod.overridePythonAttrs (
     old: {
-      propagatedBuildInputs = old.propagatedBuildInputs ++ [ pkgs.mpi ];
+      propagatedBuildInputs = old.propagatedBuildInputs ++ [ pkgs.openmpi ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix:260]:
```suggestion
      propagatedBuildInputs = old.propagatedBuildInputs ++ [ pkgs.mpi ];
```

File: pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix [hunk 3]
```diff
@@ -1317,4 +1401,47 @@ self: super:
...
+  newrelic = super.newrelic.overridePythonAttrs (
+    old: {
+      postPatch = old.postPatch or "" + ''
+        substituteInPlace setup.py --replace '"setuptools_scm>=3.2,<4"' '"setuptools_scm"'
+      '';
+    }
+  );
+
+
 }
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix:1447]:
```suggestion
  );
}
```

File: pkgs/development/tools/poetry2nix/poetry2nix/pep508.nix [hunk 4]
```diff
@@ -132,7 +132,7 @@ let
               mVal = ''[a-zA-Z0-9\'"_\. ]+'';
               mOp = "in|[!=<>]+";
               e = stripStr exprs.value;
-              m = builtins.map stripStr (builtins.match "^(${mVal}) *(${mOp}) *(${mVal})$" e);
+              m = builtins.map stripStr (builtins.match ''^(${mVal}) *(${mOp}) *(${mVal})$'' e);
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/pep508.nix:135]: Only multline strings should use ''.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
The following changes need to be reverted:
 - pkg-config -> pkgconfig
 - lib -> stdenv.lib
 - mpi -> openmpi

File: pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix (hunk 3 above)
Comment (adisbladis) [reply pkgs/development/tools/poetry2nix/poetry2nix/overrides.nix:1447]: This is auto-formatted using nixpkgs-fmt

File: pkgs/development/tools/poetry2nix/poetry2nix/pep508.nix (hunk 4 above)
Comment (adisbladis) [reply pkgs/development/tools/poetry2nix/poetry2nix/pep508.nix:135]: It doesn't matter.

**Comment by adisbladis** [comment]
@SuperSandro2000 Tbh I'm pretty livid about all those treewide PRs having gone through despite this project being entirely developed outside of nixpkgs and only periodically copied over to nixpkgs.

This is hardly the only project getting developed outside of nixpkgs and they are all ruined by these treewide PRs merge without thorough review.

**Pull request merged**